Src/DependencyCollector/Shared/DependencyTrackingTelemetryModule.cs

```diff
@@ -219,11 +224,26 @@ protected virtual void Dispose(bool disposing)
         /// <summary>
         /// Initialize for framework event source (not supported for Net40).
         /// </summary>
-        private void InitializeForFrameworkEventSource()
+        private void InitializeForDiagnosticAndFrameworkEventSource()
         {
-#if !NET40
-            FrameworkHttpProcessing frameworkHttpProcessing = new FrameworkHttpProcessing(this.telemetryConfiguration, DependencyTableStore.Instance.WebRequestCacheHolder, this.SetComponentCorrelationHttpHeaders, this.ExcludeComponentCorrelationHttpHeadersOnDomains, this.EffectiveProfileQueryEndpoint);
-            this.httpDesktopDiagnosticSourceListener = new HttpDesktopDiagnosticSourceListener(frameworkHttpProcessing);
+#if NET45
+            if (this.EnableDiagnosticSourceInstrumentation)
+            {
+                DesktopDiagnosticSourceHttpProcessing desktopHttpProcessing = new DesktopDiagnosticSourceHttpProcessing(
+                    this.telemetryConfiguration,
+                    DependencyTableStore.Instance.WebRequestConditionalHolder,
+                    this.SetComponentCorrelationHttpHeaders,
+                    this.ExcludeComponentCorrelationHttpHeadersOnDomains,
+                    this.EffectiveProfileQueryEndpoint);
+                this.httpDesktopDiagnosticSourceListener = new HttpDesktopDiagnosticSourceListener(desktopHttpProcessing);
+            }
+
+            FrameworkHttpProcessing frameworkHttpProcessing = new FrameworkHttpProcessing(
```

Comment:
IsDesktopHttpDiagnosticSourceActivated is not a runtime constant, it set to true when the DiagnosticListener is subscribed to the particular source. Potentially it could happen AFTER module is initialized. It could also be the case that particular source is not available at all, or fails to initialize.